features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/networksettled/NetworkSettledMetricResolver.kt

```diff
@@ -30,12 +33,16 @@ internal class NetworkSettledMetricResolver(
     @Volatile
     private var viewWasStopped: Boolean = false
 
+    private var currentViewDiagnostic = Diagnostic()
+
     fun viewWasCreated(eventTimestampInNanos: Long) {
         viewCreatedTimestamp = eventTimestampInNanos
+        currentViewDiagnostic = Diagnostic()
     }
 
     fun resourceWasStarted(context: InternalResourceContext) {
         if (viewWasStopped) return
+        currentViewDiagnostic.started += 1
```

Comment:
Question: networkSettledMetricResolver can be called only on a single thread, or we could have calls coming from several threads? Because some of the variables in this class are volatile, but diagnostic does not seem to be explicitly threadsafe.

Reply:
Thanks for spotting, fixed 👍